core/src/main/scala/com/microsoft/azure/synapse/ml/logging/Usage/TokenUtils.scala

```diff
@@ -0,0 +1,156 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.logging.Usage
+
+import scala.reflect.runtime.currentMirror
+import scala.reflect.runtime.universe._
+import java.time.Instant
+import org.apache.spark.SparkContext
+import com.microsoft.azure.synapse.ml.logging.SynapseMLLogging
+import spray.json.DefaultJsonProtocol.{StringJsonFormat, jsonFormat3}
+
+import java.util.UUID
+import com.microsoft.azure.synapse.ml.logging.common.WebUtils._
+
+import java.util.Date
+import pdi.jwt._
+import org.json.JSONObject
+import spray.json.RootJsonFormat
+
+import scala.util.{Failure, Success, Try}
+
+case class MwcToken (TargetUriHost: String, CapacityObjectId: String, Token: String)
+object TokenUtils {
+  var AADToken: String = ""
```

Comment:
try to avoid vars

Reply:
Required here for caching and re-populating every time the token invalidates and requires refreshing.

Reply:
Please at least make this Option instead of using empty string to code for no token